packages/mdc-menu/simple/foundation.js

```diff
@@ -411,37 +441,79 @@ class MDCSimpleMenuFoundation extends MDCFoundation {
 
     const anchor = this.adapter_.getAnchorDimensions();
     const windowDimensions = this.adapter_.getWindowDimensions();
-
-    const topOverflow = anchor.top + this.dimensions_.height - windowDimensions.height;
-    const bottomOverflow = this.dimensions_.height - anchor.bottom;
+    const topShift = (this.anchorCorner_ & CornerBit.BOTTOM) ?
```

Comment:
This entire `animationLoop_` method...can you break it up into smaller private methods? There is too much going on right now.

Reply:
Refactored... no animationLoop_ method. All is done in css. Auto-positioning logic has been broken up into smaller methods.